### PR TITLE
[AUT-4903] istio-authorizer production ready

### PR DESCRIPTION
**Pull request by ppodbielski-cloudentity**
## Description

- add config flag
- add envoyFilter template
- description update

## Type of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Tests (extending the test suite)
- [x] Refactor (internal improvement that doesn't change product functionality)
- [ ] Other (if none of the other choices apply)

### CODE REVIEW

**Comment by ppodbielski-cloudentity** [comment]
@holowinski @mbilski 

```/ $ ls
EULA.txt          dev               istio-authorizer  mnt               root              srv               usr
bin               etc               lib               opt               run               sys               var
data              home              media             proc              sbin              tmp
/ $ ls data/
ca.crt       config.yaml
/ $ ps
PID   USER     TIME  COMMAND
    1 1000      0:00 /istio-authorizer --config /data/config.yaml
   18 1000      0:00 sh
   26 1000      0:00 ps
/ $ cat data/config.yaml 
acp:
  reload_interval: 1m0s
```

File: charts/istio-authorizer/values.yaml [hunk 1]
```diff
@@ -62,35 +65,55 @@ serviceAccount:
...
+  #   cpu: 10m
+  #   memory: 48Mi
+  # limits:
+  #   cpu: 100m
+  #   memory: 96Mi
+
+## Define EnvoyFilter
+## Detailed configuration is available here
+## https://istio.io/latest/docs/reference/config/networking/envoy-filter/
+envoyFilter:
```
Comment (holowinski) [inline charts/istio-authorizer/values.yaml:80]: I think we should rename this.

envoyFilter is technically what it is, but it's not clear what's the purpose.
The main goal is to add body parsing so I would use


```
## Enable body parsing
## If you want to use request body in your authorization policies, enable this flag.
## It's going to add a filter to Envoy which will propagate the request body to the authorizer. 
parseBody:
  enabled: false
```

File: charts/istio-authorizer/values.yaml [hunk 2]
```diff
@@ -62,35 +65,55 @@ serviceAccount:
...
+  ## Enables EnvoyFilter
```
Comment (holowinski) [inline charts/istio-authorizer/values.yaml:81]: Also, this comment should be changed accordingly.

File: charts/istio-authorizer/values.yaml [hunk 3]
```diff
@@ -62,35 +65,55 @@ serviceAccount:
...
+
 ## Credentials for your ACP istio-authorizer
 ##
-##
 clientCredentials:
-  # Specifies whether a kubernetes secret should be created from the provided clientID and clientSecret.
-  # For production setup please create secret manually and use name
-  # parameter to integrate with istio-authorizer
+  ## Specifies whether a kubernetes secret should be created from the provided clientID and clientSecret.
+  ## For production setup please create secret manually and use name
```
Comment (holowinski) [inline charts/istio-authorizer/values.yaml:96]: `please create secret manually` -> `please create the secret (ideally encrypted) manually`

File: charts/istio-authorizer/values.yaml [hunk 4]
```diff
@@ -62,35 +65,55 @@ serviceAccount:
...
-## for example: root-ca: /data/ca.crt
-extraEnvVars:
-  HTTP_CLIENT_ROOT_CA: /data/ca.crt
-  # TOKEN_EXCHANGE_ENABLED: true # must be explicitly enabled for token exchange until acp 2.0 is released
-
 ## Additional istio-authorizer ConfigMap entries
 ##
 data:
-  # Example root CA used by istio-authorizer http client
+  ##  Example root CA used by istio-authorizer http client
```
Comment (holowinski) [inline charts/istio-authorizer/values.yaml:115]: Let's add
```
## If you're exposing ACP with your own custom (not publically trusted) certificates, Istio Authorizer needs to trust them.
## Set the ca.crt to the CA certificate used by ACP.
```

**Review by holowinski — CHANGES_REQUESTED** [review]

File: charts/istio-authorizer/values.yaml (hunk 1 above)
Comment (ppodbielski-cloudentity) [reply charts/istio-authorizer/values.yaml]: fixed

File: charts/istio-authorizer/values.yaml (hunk 3 above)
Comment (ppodbielski-cloudentity) [reply charts/istio-authorizer/values.yaml:90]: fixed

**Review by holowinski — APPROVED** [review]

File: charts/istio-authorizer/values.yaml [hunk 5]
```diff
@@ -62,35 +65,50 @@ serviceAccount:
...
+
 ## Credentials for your ACP istio-authorizer
 ##
-##
 clientCredentials:
-  # Specifies whether a kubernetes secret should be created from the provided clientID and clientSecret.
-  # For production setup please create secret manually and use name
-  # parameter to integrate with istio-authorizer
+  ## Specifies whether a kubernetes secret should be created from the provided clientID and clientSecret.
+  ## For production setupplease create the secret (ideally encrypted) manually
```
Comment (holowinski) [inline charts/istio-authorizer/values.yaml:90]: `setupplease` -> `setup, please`

**Review by mbilski — APPROVED** [review]

**Pull request merged**